pallets/creator-staking/src/functions.rs

```diff
@@ -0,0 +1,413 @@
+            // Ignore creator if it was unregistered
+            consumed_weight = consumed_weight.saturating_add(T::DbWeight::get().reads(1));
+            if let CreatorState::Unregistered(_) = creator_info.state {
+                continue;
+            }
+
+            // Copy data from era `X` to era `X + 1`
+            if let Some(mut staking_info) = Self::creator_stake_info(space_id, current_era)
+            {
+                staking_info.creator_reward_claimed = false;
+                CreatorEraStake::<T>::insert(space_id, next_era, staking_info);
+
+                consumed_weight =
+                    consumed_weight.saturating_add(T::DbWeight::get().reads_writes(1, 1));
+            } else {
+                consumed_weight = consumed_weight.saturating_add(T::DbWeight::get().reads(1));
+            }
+        }
+
+        consumed_weight
+    }
+
+    fn calculate_reward_for_staker_in_era(
+        creator_stake_info: &CreatorStakeInfo<BalanceOf<T>>,
+        staked: BalanceOf<T>,
+        era: EraIndex,
+    ) -> BalanceOf<T> {
+        if let Some(reward_and_stake) = Self::general_era_info(era) {
+            let (_, stakers_combined_reward_share) =
+                Self::distributed_rewards_between_creator_and_stakers(creator_stake_info, &reward_and_stake);
+            Perbill::from_rational(staked, creator_stake_info.total) * stakers_combined_reward_share
+        } else {
+            Zero::zero()
+        }
+    }
+
+    pub fn estimated_staker_rewards_by_creators(
+        staker: T::AccountId,
+        mut target_creators: Vec<SpaceId>,
+    ) -> Vec<(SpaceId, BalanceOf<T>)> {
+        let mut estimated_rewards: Vec<(SpaceId, BalanceOf<T>)> = Vec::new();
+        target_creators.dedup();
+
+        let current_era = Self::current_era();
+
+        for creator_id in target_creators {
+            let mut staker_info_for_creator = Self::staker_info(&staker, creator_id);
+
+            let mut unregistered_era = current_era;
+            if let Some(creator_info) = Self::registered_creator(creator_id) {
+                if let CreatorState::Unregistered(era) = creator_info.state {
+                    unregistered_era = era;
+                }
+            }
+
+            if staker_info_for_creator.stakes.is_empty() {
+                estimated_rewards.push((creator_id, Zero::zero()));
+                continue;
+            }
+
+            let mut total_staker_rewards_for_eras: BalanceOf<T> = Zero::zero();
+            loop {
+                let (era, staked) = staker_info_for_creator.claim();
+                if era >= unregistered_era || era == 0 {
+                    break;
+                }
+                let creator_stake_info = Self::creator_stake_info(creator_id, era).unwrap_or_default();
+
+                total_staker_rewards_for_eras = total_staker_rewards_for_eras.saturating_add(
+                    Self::calculate_reward_for_staker_in_era(&creator_stake_info, staked, era)
```

Comment:
if we have `default` value for `creator_stake_info` here, does make sense to call calculate function anyway?